helper_test.go

```diff
@@ -88,6 +89,85 @@ func fetchTestAccountDetails(t *testing.T, client *Client) *TestAccountDetails {
 	return _testAccountDetails
 }
 
+func createAgent(t *testing.T, client *Client, org *Organization, agentPool *AgentPool, agentPoolToken *AgentToken) (*Agent, *AgentPool, func()) {
+	var orgCleanup func()
+	var agentPoolCleanup func()
+	var agentPoolTokenCleanup func()
+	var agent *Agent
+
+	if org == nil {
+		org, orgCleanup = createOrganization(t, client)
+	}
+
+	upgradeOrganizationSubscription(t, client, org)
+
+	if agentPool == nil {
+		agentPool, agentPoolCleanup = createAgentPool(t, client, org)
+		t.Log("create, log agentPool: ", agentPool)
+	}
+
+	if agentPoolToken == nil {
+		agentPoolToken, agentPoolTokenCleanup = createAgentToken(t, client, agentPool)
+	}
+
+	ctx := context.Background()
+	cmd := exec.Command("docker",
+		"run", "-d",
+		"--env", "TFC_AGENT_TOKEN="+agentPoolToken.Token,
+		"--env", "TFC_AGENT_NAME="+"this-is-a-test-agent",
+		"--env", "TFC_ADDRESS="+DefaultConfig().Address,
+		"docker.mirror.hashicorp.services/hashicorp/tfc-agent:latest")
+
+	go func() {
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Logf("Could not run container: %s", err)
+		}
+
+		t.Log("Logging container output: ", (string)(output))
+	}()
+
+	defer func() {
+		t.Log("Cleaning up agent docker container: ")
+		cmd := exec.Command("docker", "rm", "-f")
+		_ = cmd.Run()
+	}()
+
+	i, err := retry(func() (interface{}, error) {
+
+		agentList, err := client.Agents.List(ctx, agentPool.ID, nil)
```

Comment:
It looks like there's an unspoken contract here: you can't ever use the same AgentPool for two concurrent executions of this function, otherwise you might get incoherent results depending on the test parallelism. Since we allow the caller to pass in their own agent pool, it might be worth stating that contract up in the function doc comment.